.github/workflows/olm_tests.yaml

```diff
@@ -0,0 +1,125 @@
+        OPERATOR_MANIFEST=deploy/operator-ci.yaml
+
+        # Create a manifest with local image
+        pushd deploy/base
+        kustomize edit set image security-profiles-operator=$IMG
+        popd
+        kustomize build --reorder=none deploy/overlays/cluster -o ${OPERATOR_MANIFEST}
+
+        # this is a kludge, we need to make sure kustomize can be overwritten
+        rm -f build/kustomize
+
+        # create bundle, bundle image, push bundle using our manifest created earlier
+        make bundle BUNDLE_OPERATOR_MANIFEST=${OPERATOR_MANIFEST} IMG=${IMG}
+        # GH CI workers have pretty limited CPU and won't be able to run SPO, OLM and cert-manager at the same time
+        sed -i '/cpu\:/d' bundle/manifests/security-profiles-operator.clusterserviceversion.yaml
+        make bundle-build BUNDLE_IMG=${BUNDLE_IMG}
+        podman push --tls-verify=false ${BUNDLE_IMG}
+
+        # create catalog image, push catalog
+        make catalog-build OPM_EXTRA_ARGS=" --skip-tls" BUNDLE_IMGS=${BUNDLE_IMG} CATALOG_IMG=${CATALOG_IMG}
+        podman push --tls-verify=false ${CATALOG_IMG}
+
+    - name: Deploy OLM
+      run: |
+        make deploy-olm
+
+    - name: Deploy SPO
+      run: |
+        REPO=localhost:5000
+        CATALOG_IMG=${REPO}/security-profiles-operator-catalog:v${GITHUB_SHA}
+
+        # cert-manager first. This should be done using dependencies in the
+        # future
+        kubectl apply -f https://github.com/jetstack/cert-manager/releases/download/v1.7.2/cert-manager.yaml
+        kubectl -ncert-manager wait --for condition=ready pod -l app.kubernetes.io/instance=cert-manager
+
+        # let's roll..
+        make deploy-using-olm CATALOG_IMG=${CATALOG_IMG}
+
+    - name: Make sure SPO is up and running
+      run: |
+        # Useful in case the CatalogSource is fubar
+        kubectl -nolm wait --for=condition=ready pods -lolm.catalogSource=security-profiles-operator
+        kubectl -nolm logs $(kubectl -nolm get pods --no-headers -lolm.catalogSource=security-profiles-operator | awk '{print $1}')
+
+        # wait a bit for CSV to appear
+        # (jhrozek): I didn't find a useful condition or status to wait for..
+        # ..if only there was a way to check if ANY installedCSV is set..
+        sleep 30
+        CSV=$(kubectl -nsecurity-profiles-operator get sub security-profiles-operator-sub -ojsonpath='{.status.installedCSV}')
+        # wait for the CSV to be actually installed
+        kubectl -nsecurity-profiles-operator wait --for=jsonpath='{.status.phase}'=Succeeded csv $CSV
+
+        # wait for the operator to be ready
+        kubectl -nsecurity-profiles-operator wait --for=condition=ready pod -lname=security-profiles-operator
+        kubectl -nsecurity-profiles-operator wait --for=condition=ready pod -lname=security-profiles-operator-webhook
+        kubectl -nsecurity-profiles-operator wait --for=condition=ready pod -lname=spod
```

Comment:
Wondering if we wanna put all of this into a dedicated script in `./hack`?

Reply:
I think at this point it would be better to have a dedicated script as @saschagrunert suggested. Let's try to keep the Makefile minimal.

Reply:
I'm fine doing the changes, but slightly confused as what exactly do you propose to put into a script in `./hack`? The whole olm action workflow? If that's what you mean, I agree very much and it would make testing of the test (heh) much easier. I'm just confused about Ozz's note that we should keep the Makefile minimal (this comment doesn't bring up the Makefile contents, but rather the contents of the ).

btw if there is a concern about the Makefile getting out of hand (which I could agree kindof) then let's split it and use something like `make -C Makefile.packaging`. I don't have very strong feelings about this though and on the other hand, I find it handy to be able to just `vi Makefile` and see all the build and packaging commands. So..opinons welcome.

Reply:
OK, I moved most of the GH action to the new bash test script, moved some one-off Makefile targets that were only used for tests there as well to keep the Makefile minimal. Hope this looks better.